goal_src/pc/pckernel.gc

```diff
@@ -131,6 +165,15 @@
     (('5120x1440) (set-size! obj 5120 1440)))
   (none))
 
+(defmethod commit-to-file pc-settings ((obj pc-settings))
+  "commits the current settings to the file"
+  ;; auto load settings if available
+  (clear *pc-temp-string-1*)
+  (format *pc-temp-string-1* "~S/pc-settings.gc" *pc-settings-folder*)
+  (write-to-file obj *pc-temp-string-1*)
+  (clear *pc-temp-string-1*)
```

Comment:
Look at the `string-format` macro in `gstring.gc`, it simplifies doing this. You can make an alternate version of it that uses that string instead of the usual `*temp-string*`.

Reply:
From what I can tell it only really saves on the calls to `clear`?  I'd rather just keeps things explicit, it's not obvious that macro uses a particular global and the pattern seem unsustainable

Reply:
You would have to just write `(write-to-file obj (pc-string-format "~S/pc-settings.gc" *pc-settings-folder*))` instead of those 4 lines.

Reply:
I understand that it saves lines, that isn't what I'm opinionated against.

Reply:
Well, the global only exists to be used a temporary string in the first place, so I don't think there's an issue.

Reply:
Someone could get into the same issue I did and it wouldn't be obvious -- they call a function using the output of `pc-string-format` and within the function it also uses `pc-string-format`.  It's not obvious what's going on unless you expand the macro.  I just prefer to be more explicit.